homeassistant/components/template/coordinator.py

```diff
@@ -91,16 +101,68 @@ async def _attach_triggers(self, start_event=None) -> None:
             start_event is not None,
         )
 
+    async def _setup_condition_checker(
+        self,
+    ) -> Callable[[TemplateVarsType | None], bool] | None:
+        cond_configs = self.config[CONF_CONDITION]
+        checks: list[condition.ConditionCheckerType] = [
+            await condition.async_from_config(self.hass, cond_config)
+            for cond_config in cond_configs
+        ]
+
+        def check_conditions(variables: TemplateVarsType | None = None) -> bool:
+            """AND all conditions."""
+            errors: list[ConditionErrorIndex] = []
+            for index, check in enumerate(checks):
+                try:
+                    with trace_path(["condition", str(index)]):
+                        if check(self.hass, variables) is False:
+                            return False
+                except ConditionError as ex:
+                    errors.append(
+                        ConditionErrorIndex(
+                            "condition", index=index, total=len(checks), error=ex
+                        )
+                    )
+
+            if errors:
+                _LOGGER.warning(
+                    "Error evaluating template trigger condition:\n%s",
+                    ConditionErrorContainer("condition", errors=errors),
+                )
+                return False
+
+            return True
+
+        return check_conditions
```

Comment:
This code is copied from `homeassistant/components/automation/__init__.py`; we should instead move the common code to the condition helper.

Reply:
Deduplicated in 2d9928d